src/Data/Base64/Internal.hs

```diff
@@ -0,0 +1,26 @@
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE DataKinds #-}
+
+module Data.Base64.Internal(Alphabet(..), Base64(..), mapBase64) where
+
+-- | The different kinds of supported Base64 encodings
+data Alphabet
+    -- | Standard base64 according to [RFC 4648 §4](https://datatracker.ietf.org/doc/html/rfc4648#section-4)
+    -- Padding is always inserted when encoding, and required when decoding
+    = StdPadded
+    -- | Standard base64 according to [RFC 4648 §4](https://datatracker.ietf.org/doc/html/rfc4648#section-4)
+    -- Padding is never inserted when encoding, and optional when decoding
+    | StdUnpadded
+    -- | URL-safe base64 according to [RFC 4648 §5](https://datatracker.ietf.org/doc/html/rfc4648#section-5) aka base64url
+    -- Padding is always inserted when encoding, and required when decoding
+    | UrlPadded
+    -- | URL-safe base64 according to [RFC 4648 §5](https://datatracker.ietf.org/doc/html/rfc4648#section-5) aka base64url
+    -- Padding is never inserted when encoding, and optional when decoding
+    | UrlUnpadded
```

Comment:
I don't think we'll need the `'StdUnpadded` for the std alphabet, since we assume that all standard alphabet base64 encoded strings are padded, with the lenient decoders provided as a convenience for when you have concatenated base64 strings separated by some non-alphabet char (which is used in some apis). 

The URL-safe alphabet is the only one that's specified to be fuzzy on what it accepts, and has the option for padding.

Reply:
Resolved in https://github.com/emilypi/base64/commit/e2bdbe14244b7542a4d68885ce41ead426dccf54